python/ffsim/qiskit/sim.py

```diff
@@ -715,6 +726,29 @@ def _evolve_state_vector_spinful(
         )
         return states.StateVector(vec=vec, norb=norb, nelec=nelec)
 
+    if isinstance(op, PermutationGate):
+        perm = list(op.pattern)
+        qs = qubit_indices
+        inverse_perm = [0] * len(perm)
+        for out_idx, src_idx in enumerate(perm):
+            inverse_perm[src_idx] = out_idx
```

Comment:
If `perm` can remain a Numpy array, then the inverse permutation can be computed more simply like this:
```python
        inverse_perm = np.empty_like(perm)
        inverse_perm[perm] = np.arange(len(perm))
```